pkg/component/runtime/runtime.go

```diff
@@ -187,6 +187,10 @@ func (s *componentRuntimeState) start() error {
 }
 
 func (s *componentRuntimeState) stop(teardown bool, signed *component.Signed) error {
+	if s.shuttingDown.Load() {
+		// already stopping
+		return nil
+	}
```

Comment:
Can `shuttingDown` be set to false between this `Load` and the following `Store`?

Reply:
from a component point of view it should be guarded, but from agent architecture standpoint it should not be concurrent and should be safe to have it as it is